openrouteservice/src/main/java/org/heigit/ors/api/converters/APIRequestProfileConverter.java

```diff
@@ -15,17 +15,23 @@
 
 package org.heigit.ors.api.converters;
 
-import org.springframework.core.convert.converter.Converter;
 import org.heigit.ors.api.requests.common.APIEnums;
+import org.springframework.core.convert.converter.Converter;
 
 public class APIRequestProfileConverter implements Converter<String, APIEnums.Profile> {
     @Override
+    @SuppressWarnings("java:S112")
     public APIEnums.Profile convert(String s) {
-        for(APIEnums.Profile profile : APIEnums.Profile.values()) {
-            if(profile.toString().equals(s)) {
+        for (APIEnums.Profile profile : APIEnums.Profile.values()) {
+            if (profile.toString().equals(s)) {
                 return profile;
             }
         }
-        return null;
+        // This is a workaround for the breaking change: https://github.com/spring-projects/spring-framework/issues/26679
+        // This is fixed with a "missingAfterConversion" flag in spring-boot 3.x.x.
+        // Required parameters raised a HttpMessageConversionException before spring 2.7.x. when present but wrong.
+        // With spring boot 2.7.x it raises a MissingServletRequestParameterException.
+        // By throwing a RuntimeException in the custom converter spring itself now raises a HttpMessageConversionException.
+        throw new RuntimeException("");
```

Comment:
Try to be more specific with the exception, even in a workaround:
```suggestion
throw new IllegalArgumentException("unsupported profile");
```